### ISSUE TITLE
Error decoding Date Strings when setting `dateDecodingStrategy = .iso8601` on a `Gettable`

### ISSUE THREAD

**Issue by adealmeida** [ryanbaldwin/Restivus#8]
I've created the following request with corresponding response structure:

```
struct AqhiForecastContextual: Decodable {
    var airQualityHealthIndex: Double
    var periodName: String
    var periodValue: String
}

struct AqhiForecastHourly: Decodable {
    var airQualityHealthIndex: Double
    var associatedRisk: String
    var timestamp: Date
}

struct AqhiForecastResponse: Decodable {
    var siteId: String?
    var regionName: String?
    var forecastContextual: [AqhiForecastContextual]?
    var forecastHourly: [AqhiForecastHourly]?
}

struct AqhiForecastRequest {
    var latitude: Double
    var longitude: Double
}

extension AqhiForecastRequest: Interceptable, Gettable {
    public typealias ResponseType = AqhiForecastResponse
    public var path: String { return "/api/environmental/AQHI/forecast?latitude=\(latitude)&longitude=\(longitude)" }
    public var dateDecodingStrategy: JSONDecoder.DateDecodingStrategy { return .formatted(DateFormatter.iso8601Extended) }
}
```

When the request is submitted, it receives the following JSON response from the server:

```
Response:
===============
Response Code: 200
Headers:
	Content-Type: application/json;charset=UTF-8
	Connection: close
	Transfer-Encoding: Identity
	Server: Apache-Coyote/1.1
	Date: Tue, 24 Apr 2018 19:40:22 GMT
Data: 

{
  "siteId" : "FCWYG",
  "regionName" : "Toronto Downtown",
  "forecastContextual" : [ {
    "airQualityHealthIndex" : 5,
    "periodName" : "Today",
    "periodValue" : "Tuesday",
    "periodId" : 1
  }, {
    "airQualityHealthIndex" : 4,
    "periodName" : "Tonight",
    "periodValue" : "Tuesday night",
    "periodId" : 2
  }, {
    "airQualityHealthIndex" : 4,
    "periodName" : "Tomorrow",
    "periodValue" : "Wednesday",
    "periodId" : 3
  } ],
  "forecastHourly" : [ {
    "airQualityHealthIndex" : 4.0,
    "associatedRisk" : "MODERATE",
    "timestamp" : "2018-04-24T10:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 4.0,
    "associatedRisk" : "MODERATE",
    "timestamp" : "2018-04-24T11:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 5.0,
    "associatedRisk" : "MODERATE",
    "timestamp" : "2018-04-24T12:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 4.0,
    "associatedRisk" : "MODERATE",
    "timestamp" : "2018-04-24T13:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 4.0,
    "associatedRisk" : "MODERATE",
    "timestamp" : "2018-04-24T14:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 4.0,
    "associatedRisk" : "MODERATE",
    "timestamp" : "2018-04-24T15:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 4.0,
    "associatedRisk" : "MODERATE",
    "timestamp" : "2018-04-24T16:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 4.0,
    "associatedRisk" : "MODERATE",
    "timestamp" : "2018-04-24T17:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 3.0,
    "associatedRisk" : "LOW",
    "timestamp" : "2018-04-24T18:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 3.0,
    "associatedRisk" : "LOW",
    "timestamp" : "2018-04-24T19:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 3.0,
    "associatedRisk" : "LOW",
    "timestamp" : "2018-04-24T20:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 4.0,
    "associatedRisk" : "MODERATE",
    "timestamp" : "2018-04-24T21:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 4.0,
    "associatedRisk" : "MODERATE",
    "timestamp" : "2018-04-24T22:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 4.0,
    "associatedRisk" : "MODERATE",
    "timestamp" : "2018-04-24T23:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 4.0,
    "associatedRisk" : "MODERATE",
    "timestamp" : "2018-04-25T00:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 4.0,
    "associatedRisk" : "MODERATE",
    "timestamp" : "2018-04-25T01:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 4.0,
    "associatedRisk" : "MODERATE",
    "timestamp" : "2018-04-25T02:00:00.000-04:00"
  }, {
    "airQualityHealthIndex" : 3.0,
    "associatedRisk" : "LOW",
    "timestamp" : "2018-04-25T03:00:00.000-04:00"
  } ]
}
```

When trying to decode the JSON into the response struct, the following error is output:

```
typeMismatch(Swift.Double, Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "forecastHourly", intValue: nil), _JSONKey(stringValue: "Index 0", intValue: 0), CodingKeys(stringValue: "timestamp", intValue: nil)], debugDescription: "Expected to decode Double but found a string/data instead.", underlyingError: nil))
```

**Comment by ryanbaldwin** [comment]
Looking into this now! Thanks Adrian!

**Comment by ryanbaldwin** [comment]
This is now fixed and will be part of 0.4.2, which I'll draft a release for immediately.
